pkg/entity/utilities/document.go

```diff
@@ -162,25 +176,36 @@ func ToGameHistory(doc *ipc.GameDocument, cfg *config.Config) (*macondo.GameHist
 				rack = doc.Events[index+1].Rack
 			}
 		}
+		var tilesFromRack int
+		if evt.Type == ipc.GameEvent_EXCHANGE {
+			tilesFromRack = len(evt.Exchanged)
```

Comment:
does this handle unicode tiles?

Reply:
it does actually. This event is a `GameDocument` event (`GameEvent`) not a Macondo `GameEvent`. The latter only deals with strings, the former deals with bytes and "`MachineLetters`".